### PR TITLE
[Merged by Bors] - Remove NipostChallenge from `ActivationTxHeader`

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

Simplify the `ActivationTxHeader` struct by removing NipostChallenge and a few unused fields.

## Description

Additionally, Atx Builder doesn't depend on CachedDB anymore.

## Test Plan

Existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Review by fasmat — APPROVED** [review]

File: activation/activation.go [hunk 1]
```diff
@@ -463,10 +462,10 @@ func (b *Builder) BuildNIPostChallenge(ctx context.Context, nodeID types.NodeID)
 		return challenge, nil
 	}
 
-	prev, err := b.cdb.GetLastAtx(nodeID)
+	prevAtx, err := b.GetPrevAtx(nodeID)
```
Comment (ivan4th) [inline activation/activation.go:465]: As this `CachedDB.GetLastAtx` isn't used in other places, perhaps it is worth removing altogether?

**Review by ivan4th — CHANGES_REQUESTED** [review]

**Comment by fasmat** [comment]
In the future we should get rid of ActivationTxHeader completely and replace it with the domain type ATX

File: activation/activation.go (hunk 1 above)
Comment (poszu) [reply activation/activation.go:465]: Removed it and a few others unused.

**Review by ivan4th — APPROVED** [review]

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
Multiple errors that seem to be related to the environment rather than the change. bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
Found and fixed the problem causing systests to fail - bors merge

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors cancel

**Comment by poszu** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**